packages/terra-functional-testing/src/services/wdio-visual-regression-service/VisualRegressionLauncher.js

```diff
@@ -63,7 +64,30 @@ class VisualRegressionLauncher {
    * Function to be executed after a test in Mocha.
    * @param {Object} test - test details
    */
-  afterTest() {
+  afterTest(test, _, { passed }) {
+    const { config } = browser;
+    const { formFactor } = config.launcherOptions;
+    const screenshotContext = {
+      desiredCapabilities: this.context.desiredCapabilities,
+      suite: this.currentSuite,
+      test: this.currentTest,
+      meta: {
+        currentFormFactor: formFactor,
+      },
+      options: {
+        name: test.title,
+      },
+    };
+
+    const screenshotContextCleaned = lodashPickby(screenshotContext, lodashIdentity);
+    const { errorPath } = this.compare.getScreenshotPaths(screenshotContextCleaned);
+    if (passed) {
+      fse.removeSync(errorPath);
+    } else {
+      fse.ensureFileSync(errorPath);
+      browser.saveScreenshot(errorPath);
+    }
```

Comment:
What does the screenshot get named?

Reply:
These screenshots are named after the name of the test. Since these would also be generated for tests that don't necessarily have one of our matchers with a name defined, I thought that was likely the best bet for a meaningful name instead of "Error_{timestamp}", might be a little overkill because the test name ends up creating subdirectories for the suite name, then any describe blocks. I could try splitting it to just get the final 'it' block name and adding a timestamp for ensured uniqueness.

Reply:
The it-block-name_timestamp makes sense to me, although it might result in a long filename. Whichever naming convention we choose I don't think it should be creating any additional directories.

Does the test object have a `fullName` property? This should be a single string that concats the suites, describes, and test title together

Reply:
I thought that the test object would have the `fullTitle`, but for some reason when I was testing it was always undefined at this point.

Reply:
You can try `test.parent + ' ' + test.title`. See https://github.com/webdriverio/webdriverio/issues/5373#issuecomment-625267449